src/enkf_chgres_recenter_nc.fd/CMakeLists.txt

```diff
@@ -8,7 +8,10 @@ utils.f90
 )
 
 if(CMAKE_Fortran_COMPILER_ID MATCHES "^(Intel)$")
-  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fp-model precise")
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fp-model precise -fpp")
+endif()
+if(ip_VERSION GREATER_EQUAL 4.0.0)
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -DIP_V4")
```

Comment:
I think `target_compile_definitions(exe_name PRIVATE  "IP_V4")` should be used before the `target_link_libraries` line below.

You shouldn't then need the Fortran preprocessor (`-fpp`) compiler option.

Reply:
@aerorahul When I try this I receive the cmake error
```
CMake Error at src/enkf_chgres_recenter_nc.fd/CMakeLists.txt:19 (target_compile_definitions):
  Cannot specify compile definitions for target "exe_name" which is not built
  by this project.
```

Reply:
Oh, I see it.  It should be ${exe_name}.

Reply:
yeah. I updated my lazy comment.

Reply:
But using this seems to dis-envoke the preprocessor without `-fpp`:

/scratch1/NCEPDEV/nems/David.Huber/GW/gw_spack-stack_151/sorc/gfs_utils.fd/src/enkf_chgres_recenter_nc.fd/interp.f90(210): warning #5117: Bad # preprocessor line
```
#ifdef IP_V4
-^
/scratch1/NCEPDEV/nems/David.Huber/GW/gw_spack-stack_151/sorc/gfs_utils.fd/src/enkf_chgres_recenter_nc.fd/interp.f90(212): warning #5117: Bad # preprocessor line
#endif
```

Reply:
Hmm.  I have never had that problem and the documentation doesn't seem to mention this:
https://cmake.org/cmake/help/latest/command/target_compile_definitions.html#command:target_compile_definitions

Reply:
I know.  Rename the source code to `.F90`.

Reply:
CMake will invoke the preprocessor automatically if the extension is `.F90` to create the `.f90` file prepared for the compiler.